src/modality-specific-files/physiological-recordings.md

```diff
@@ -66,118 +164,639 @@ Additional metadata may be included as in
+  "sub-01": {
+    "func": {
+      "sub-01_task-nback_physio.json": "",
+      "sub-01_task-nback_physio.tsv.gz": "",
+      "sub-01_task-nback_physioevents.json": "",
+      "sub-01_task-nback_physioevents.tsv.gz": "",
+    },
+  },
+}) }}
+
+However, the `sub-01_task-nback_physio.json` will define now an extra
+column called `"timestamp"`:
 
 ```JSON
 {
     "SamplingFrequency": 100.0,
     "StartTime": -22.345,
-    "Columns": ["cardiac", "respiratory", "trigger"],
-    "Manufacturer": "Brain Research Equipment ltd.",
+    "Columns": ["timestamp", "cardiac"],
     "cardiac": {
         "Description": "continuous pulse measurement",
         "Units": "mV"
-        },
-    "respiratory": {
-        "Description": "continuous measurements by respiration belt",
-        "Units": "mV"
-        },
-    "trigger": {
-        "Description": "continuous measurement of the scanner trigger signal"
-        }
+    },
+    "timestamp": {
+        "Description": "a continuously increasing identifier of the sampling time registered by the device",
+        "Units": "ms",
+        "Origin": "System startup"
+    }
 }
 ```
 
-Note how apart from the general metadata fields like `SamplingFrequency`, `StartTime`, `Columns`,
-and `Manufacturer`,
-each individual column in the TSV file may be documented as its own field in the JSON file
-(identical to the practice in other TSV+JSON file pairs).
-Here, only the `Description` and `Units` fields are shown, but you may use any other of the
-[defined fields](../common-principles.md#tabular-files) such as `TermURL`, `LongName`, and so on.
-In this example, the `"cardiac"` and `"respiratory"` time series are produced by devices from
-the same manufacturer and follow the same sampling frequency.
-To specify different sampling frequencies or manufacturers, the time series would have to be split
-into separate files like `*_recording-cardiac_physio.<tsv.gz|json>` and `*_recording-respiratory_physio.<tsv.gz|json>`.
+The decompressed contents of `sub-01_task-nback_physio.tsv.gz`
+are now:
 
-## Recommendations for specific use cases
+```{.text noheader="1" linenums="1"}
+10.1	13894432329
+10.0	13894432330
+9.5	13894432331
+9.2	13894432332
+9.0	13894432333
+10.2	13894432334
+10.3	13894432335
+10.1	13894432336
+```
 
-To store pulse or breathing measurements, or the scanner trigger signal, the
-following naming conventions SHOULD be used for the column names:
+Then, the `sub-01_task-nback_physioevents.json` MAY define
+the `ForeignIndexColumn` for indexing:
 
-<!-- This block generates a columns table.
+```JSON
+{
+    "Columns": ["onset", "message"],
+    "Description": "Messages logged by the measurement device",
+    "ForeignIndexColumn": "timestamp"
+}
+```
+
+The decompressed contents of the corresponding TSV file,
+`sub-01_task-nback_physioevents.tsv.gz` could read now:
+
+```tsv
+13894432325	Ready
+13894432331	Synchronous recalibration triggered
+13894432334	External message received: new block
+```
+
```

Comment:
I'll note that the onset of the "Ready" message is in the scale of `timestamp` column, but is not a value explicitly present in that column.

That's fine, but might be worth saying explicitly that `onset` values may not correspond to exact values in the `ForeignIndexColumn`. And there will be little the validator can do on this point apart from enforce the same type in the two columns.